### ISSUE TITLE
A few 3D plot issues

### ISSUE THREAD

**Issue by joa-quim** [GenericMappingTools/gmt#4699]
I'm trying to reproduce the last Matlab fig in this [blog ](https://www.mathworks.com/matlabcentral/fileexchange/49813-matlab-for-analyzing-and-visualizing-geospatial-data) but a few things are not good. Note how the labels are still plotted even behind the *walls*. Not easy to see though because due to low elevation angle the labels and annotations are nearly invisible. We should have an option like in Matlab to plot them (annots & labels) without perspective. An option to have a vertical colorscale would be nice too.

Also, the warning
> psscale [WARNING]: -R with six parameters but no -Jz|Z given - ignore zmin/zmax

is a generic one (not made for psscale) but annoying anyway and hard to avoid from generated code. Better to simply shut up in these cases.

And we have the problem of the symbols in 3D. Circles won't work. There is that postscript code for spheres but a possible simpler solution is to simulate the spheres as circles without perspective.

```
julia> scatter3(xEast./1000,yNorth./1000,zDepth./1000, ms=ptSz, view=(200,5), zcolor=eqsJ.mw, colorbar=true, aspect=:equal, xlabel="East (km)", ylabel="North (km)", zlabel="Depth (km)", JZ=5, marker=:u, axes="afg zafg", fmt=:png, show=1, Vd=1)
        ["psxyz  -JX12c/0 -Bafg -Bzafg -Bx+l\"East (km)\" -By+l\"North (km)\" -Bz+l\"Depth (km)\" -JZ5 -p200/5 -R-920/20/-340/240/-425.56697809/-11.1037594056 -C -Su -P -K > C:\\TMP\\GMTjl_tmp.ps", "psscale  -R -J -Baf -p200/5 -DJMR -C -K -O >> C:\\TMP\\GMTjl_tmp.ps"]
psscale [WARNING]: -R with six parameters but no -Jz|Z given - ignore zmin/zmax
```

![GMTjl_tmp](https://user-images.githubusercontent.com/537321/105650961-67472100-5ead-11eb-879b-645bb0ece204.png)

**Comment by joa-quim** [comment]
One more thing that would be nice to have changed. Notice how the thickness of the annotation axis changes depending on the perspective. Would look better if it has always the same thickness like in the Matlab fig.

**Comment by PaulWessel** [comment]
You could paint the walls white if you don't want to determine which -B settings are required.  As for spheres, yes I think we want that.  Ideally ellipsoids but I have only seen spheres with illumination in PS.
Plotting annotations without perspective is no longer simple I think after Remko is doing it all via a matrix transformation of a plan.  In the old days we did the 3-D perspective of text ourselves but now the whole plane is done that way. Maybe there is a way to plot them in the xz and yz planes instead when the elevation is < some small angle like here.
I do not see their equivalent plot in your link thought.

**Comment by joa-quim** [comment]
Plotting in the vertical planes is probably what Ml does.
![image](https://user-images.githubusercontent.com/537321/105724310-9eeeb100-5f1f-11eb-910b-6ff138a76292.png)

**Comment by PaulWessel** [comment]
No, Matlab only plots text straight on the page, no perspective ever.  It is always like this as far as I can remember.  Perhaps we can entice @remkos to offer some thoughts on how we would go about plotting the x,y labels in the xz and yz planes when the elevation angle is so low, say less than **MAP_ELEVATION_CUTOFFWHATEVER** [25].

**Comment by jmbarbagelata** [comment]
Dear autors, if you think in include spheres and ellipsoids in plot3d, you maybe could also include 3D focal mechanism beach ball symbols. All this thigs would be so usefull.

**Comment by PaulWessel** [comment]
Agreed, but note these are very hard things to draw in a vector-based language such as PostScript.  We have plans to add shaded spheres and possibly ellipsoids.  Beach balls may have to wait.

**Comment by joa-quim** [comment]
WE have this already (except 3D beach-balls)
